metric2/asyncmetric/asyncfloat64metric/asyncfloat64metric.go

```diff
@@ -0,0 +1,53 @@
+package asyncfloat64metric
```

Comment:
What is the reason for such fine-grained packages separation? What problem does it solve? Would it not make the API less discoverable?

Reply:
I was experimenting, here. @rakyll gave feedback a while back, [here](https://docs.google.com/document/d/1WQ9_KQTAu2cfkh4A0OZFB9DfEtZkBa-1x10JuQJK5mM/edit), which suggested focusing on the package directory structure and ensuring that the top-level package is a good entry point. 

The problem I am trying to address is that we have 2 number types, and 2 calling styles. The number types create parallel interfaces that are essentially identical, and the calling styles create parallel interfaces with certain differences. If you put all four of these interfaces into one package, it becomes difficult to read. My expectation is that users will not have to import these sub-packages, they would declare instruments like:

```
   myCounter = metricglobal.Meter("library").Synchronous().Counter("name1")
   myGauge = metricglobal.Meter("library").Asynchronous().Gauge("name2")
```

Reply:
Just giving a few cents (not sure how much help it would be). e.g. [`github.com/prometheus/client_golang/prometheus`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#hdr-Metrics) has the following description to address the overwhelming godoc:

> The number of exported identifiers in this package might appear a bit overwhelming. However, in addition to the basic plumbing shown in the example above, you only need to understand the different metric types and their vector versions for basic usage.

Not an easy problem to address...

Reply:
Would it be prudent to remove some of the repetition?
Like:
- `metric2/asyncmetric/float64/asyncfloat64metric.go`
- `metric2/syncmetric/float64/asyncfloat64metric.go`

Reply:
I like this, will try it out.